src/app/CommandSender.cpp

```diff
@@ -61,6 +62,28 @@ CHIP_ERROR CommandSender::AllocateBuffer()
     return CHIP_NO_ERROR;
 }
 
+#if CONFIG_BUILD_FOR_HOST_UNIT_TEST
+CHIP_ERROR CommandSender::TestOnlyCommandSenderTimedRequestFlagWithNoTimedInvoke(const SessionHandle & session,
+                                                                                 Optional<System::Clock::Timeout> timeout)
+{
+    VerifyOrReturnError(mTimedRequest, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(mState == State::AddedCommand, CHIP_ERROR_INCORRECT_STATE);
```

Comment:
Everything after the first line of this method seems like it duplicates `CommandSender::SendCommandRequest` except for the `mTimedRequest != mTimedInvokeTimeoutMs.HasValue()` bit in that function.

Either here or in a followup, can we please have these mTimedRequest sanity-checks up front (and this method should probably also test that `!mTimedInvokeTimeoutMs.HasValue()`?), and then call a shared helper for the exchange-creation and whatnot instead of duplicating that bit.

Reply:
Updated in the most recent commit.